### ISSUE TITLE
Bad encoding in query parameters with new TestClient using httpx.Client

### ISSUE THREAD

**Issue by Kludex** [Kludex/starlette#1952]
### Discussed in https://github.com/encode/starlette/discussions/1946

<div type='discussions-op-text'>

<sup>Originally posted by **azogue** November 15, 2022</sup>
After the change to `httpx` for the `TestClient` in **v0.21.0** (), the query parameters are not properly encoded? when sending requests with it, and **strings are corrupted** when received in the endpoints.

Quick example to show the problem:

```python
import logging
from fastapi import FastAPI

app = FastAPI()

@app.get("/example")
async def _show_encoding_error(look_for: str):
    return {"found": look_for}

if __name__ == '__main__':
    from starlette.testclient import TestClient

    with TestClient(app) as client:
        params = {"look_for": "plain text"}
        resp = client.get("/example", params=params).json()
        logging.warning(resp)
        assert resp["found"] == "plain text"

        params = {"look_for": "España"}
        resp = client.get("/example", params=params).json()
        logging.warning(resp)
        assert resp["found"] == "España", resp["found"]
```

The example app works as expected if called from the SwaggerUI or from another python process using a plain `httpx.Client`, so it appears something broke with the new wrapping for `TestClient` 🥲

```python
import httpx

params = {"look_for": "España"}
with httpx.Client(base_url="http://localhost:8000/") as client:
    resp = client.get("/example", params=params).json()
    assert resp["found"] == "España"
```

#### Details

First issued in fastapi repo: https://github.com/tiangolo/fastapi/issues/5646, before checking the `TestClient` comes from here 🙈

Environment:
- macOS M1 arm64
- fastapi v0.87.0
- starlette v0.21.0
- httpx v0.23.0

</div>

**Comment by azogue** [comment]
I just checked that removing both unquote operations for the netloc and the query parts does work as expected and does not break any of the tests 🥳 (609 passed, 2 xfailed in 2.41s in quick venv with py3.10)

So it could be as simple as that to fix this 🤩

**Comment by Kludex** [comment]
This was fixed on #1953 , and it will be available on Starlette 0.22.0.
